clicache/src/Cache.cpp

```diff
@@ -339,7 +339,8 @@ namespace Apache
 
         PoolFactory^ Cache::GetPoolFactory()
         {
-          return PoolFactory::Create(m_nativeptr->get_shared_ptr()->getPoolManager().createFactory());
+          return PoolFactory::Create(std::make_shared<native::PoolFactory>(
```

Comment:
The .NET factory does not need to keep a shared pointer to the C++ factory. It is a 1:1 relationship now so it should keep a unique_ptr. See other factories that have gone through this change for examples.

Reply:
Thanks. Agreed that unique_ptr is better here.